### ISSUE TITLE
Properties ["groups"] outside of the schema when none are allowed

### ISSUE THREAD

**Issue by joakim-sch** [dependabot/dependabot-core#7897]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

npm

### dependabot.yml content

```
version: 2
      
registries: 
  artifactory:
    type: npm-registry
    url: ***
    username: ${{ secrets.ARTIFACTORY_USERNAME }}
    password: ${{ secrets.ARTIFACTORY_PASSWORD }}
    
updates:
  - package-ecosystem: "npm"
    registries:
      - artifactory
    directory: "/"
    schedule:
      interval: "weekly"
      day: "monday"
      time: "05:30"
      timezone: "Europe/Copenhagen"
    groups:
      wdio:
        patterns:
          - "@wdio*"
```

### What you expected to see, versus what you actually saw

Expected dependabot to group dependencies, but instead it says that;

The property '#/updates/0/' contains additional properties ["groups"] outside of the schema when none are allowed

**Comment by jakecoffman** [comment]
I copied the dependabot.yml into a [test repo](https://github.com/dsp-testing/npm-group-config-failure) and it validated ok.

Maybe double check the whitespaces are all consistent?

**Comment by joakim-sch** [comment]
Thank you very much for the reply @jakecoffman, and for testing it in your test-repo 😄 
I do see that it works for you in a public repo.
Unfortunately I'm not having any luck. My repo is in our organizations GH Enterprise and so I cannot share it, but despite trying to copy the raw file from your repo into mine, and even downloaded and replacing the dependabot.yml with yours altogether, I still get the error message shown below.
![image](https://github.com/dependabot/dependabot-core/assets/97220950/5a60c3bf-cd47-4bef-9922-7431326eb414)

Could it be something like this functionality not being available in GH Enterprise?
I read [here](https://github.blog/changelog/2023-06-30-grouped-version-updates-for-dependabot-public-beta/) that its a public beta, maybe it's not available to enterprise customers yet?

**Comment by jurre** [comment]
@joakim-sch is your org using a self-hosted GH Enterprise instance, or are you using github.com?

**Comment by joakim-sch** [comment]
> @joakim-sch is your org using a self-hosted GH Enterprise instance, or are you using github.com?

Self-hosted

**Comment by GandalfIX** [comment]
Hey there,
I have the same issue in a self-hosted GH-enterprise environment. I actually copied the angular example from the documentation page and I get the same error as described above.

We are running on "GitHub Enterprise Server 3.8.5"

Any additional tip would be highly appreciated :-)

`
version: 2
updates:
  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "weekly"
    target-branch: "main"
    versioning-strategy: "lockfile-only"
    allow:
      - dependency-type: "direct"
    open-pull-requests-limit: 10
    groups:
      angular:
        patterns:
          - "@angular*"
        update-types:
          - "minor"
          - "patch"

`

**Comment by keithMorgadoRakuten** [comment]
Same error here, grouping not working on Enterprise.

**Comment by andrewbcoyle** [comment]
Same issue here.

**Comment by Nishnha** [comment]
Hi, grouped updates are only available in GHES 3.11 at the moment - https://docs.github.com/en/enterprise-server@3.11/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups
